### ISSUE TITLE
pylint complains on import of "_" from ibis

### ISSUE THREAD

**Issue by rhstanton** [pylint-dev/pylint#10096]
### Bug description

```python
The code below runs fine, but pylint reports the error "No name '_' in module 'ibis' [no-name-in-module] against the line starting "from ibis...". 

-------------------

"""Test code"""

import ibis
from ibis import _
import pandas as pd

ibis.options.interactive = True

df = ibis.memtable(
    pd.DataFrame(
        [["a", 1, 2], ["b", 3, 4]],
        columns=["one", "two", "three"],
        index=[5, 6],
    )
)

df.select(_.one, _.two)
```

### Configuration

```ini

```

### Command used

```shell
pylint ibis_test.py
```

### Pylint output

```python
************* Module ibis_test
ibis_test.py:4:0: E0611: No name '_' in module 'ibis' (no-name-in-module)
```

### Expected behavior

This is valid code, so I'd expect no error

### Pylint version

```shell
pylint 3.2.7
astroid 3.2.4
Python 3.12.2 | packaged by conda-forge | (main, Feb 16 2024, 20:54:21) [Clang 16.0.6 ]
```

### OS / Environment

macOS 15.1.1 running on an Apple chip. Using Anaconda Python.

### Additional dependencies

```python

```
